### PR TITLE
Mark locality management types and operations unstable

### PR DESCRIPTION
**Pull request by OlivierHecart**

### CODE REVIEW

File: zenoh/src/lib.rs [hunk 1]
```diff
@@ -227,10 +227,10 @@ where
 {
     OpenBuilder {
         config,
-        publications_destination: Locality::Any,
-        subscribers_origin: Locality::Any,
-        queries_destination: Locality::Any,
-        queryables_origin: Locality::Any,
+        publications_destination: Locality::default(),
```
Comment (Mallets) [inline zenoh/src/lib.rs:230]: `Locality` should be only on subscriber and queryable side for the time being at least.

File: zenoh/src/lib.rs [hunk 2]
```diff
@@ -227,10 +227,10 @@ where
...
+        subscribers_origin: Locality::default(),
+        queries_destination: Locality::default(),
```
Comment (Mallets) [inline zenoh/src/lib.rs:232]: `Locality` should be only on subscriber and queryable side for the time being at least.

File: zenoh/src/lib.rs [hunk 3]
```diff
@@ -264,6 +264,7 @@ where
 {
     /// Restrict the matching subscribers that will receive data published from this [`Session`]
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:267]: `Locality` should not be configurable at session level but only per `declare_subscriber` and `declare_queryable`-basis.

File: zenoh/src/lib.rs [hunk 4]
```diff
@@ -272,13 +273,15 @@ where
 
     /// Restrict the matching publications that will be receive by the [`Subscribers`](crate::subscriber::Subscriber)
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:276]: `Locality` should not be configurable at session level but only per `declare_subscriber` and `declare_queryable`-basis.

File: zenoh/src/lib.rs [hunk 5]
```diff
@@ -287,6 +290,7 @@ where
 
     /// Restrict the matching queries that will be receive by the [`Queryables`](crate::queryable::Queryable)
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:293]: `Locality` should not be configurable at session level but only per `declare_subscriber` and `declare_queryable`-basis.

File: zenoh/src/lib.rs [hunk 6]
```diff
@@ -356,10 +360,10 @@ where
 pub fn init(runtime: Runtime) -> InitBuilder {
     InitBuilder {
         runtime,
-        publications_destination: Locality::Any,
-        subscribers_origin: Locality::Any,
-        queries_destination: Locality::Any,
-        queryables_origin: Locality::Any,
+        publications_destination: Locality::default(),
```
Comment (Mallets) [inline zenoh/src/lib.rs:363]: `Locality` should be only on subscriber and queryable side for the time being at least.

File: zenoh/src/lib.rs [hunk 7]
```diff
@@ -356,10 +360,10 @@ where
...
+        subscribers_origin: Locality::default(),
+        queries_destination: Locality::default(),
```
Comment (Mallets) [inline zenoh/src/lib.rs:365]: Locality should be only on subscriber side for the time being at least.

File: zenoh/src/lib.rs [hunk 8]
```diff
@@ -380,6 +384,7 @@ pub struct InitBuilder {
 impl InitBuilder {
     /// Restrict the matching subscribers that will receive data published from this [`Session`]
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:387]: Locality should not be configurable at session level but only per declare_subscriber and declare_queryable-basis.

File: zenoh/src/lib.rs [hunk 9]
```diff
@@ -388,13 +393,15 @@ impl InitBuilder {
 
     /// Restrict the matching publications that will be receive by the [`Subscribers`](crate::subscriber::Subscriber)
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:396]: Locality should not be configurable at session level but only per declare_subscriber and declare_queryable-basis.

File: zenoh/src/lib.rs [hunk 10]
```diff
@@ -388,13 +393,15 @@ impl InitBuilder {
...
     #[inline]
     pub fn subscribers_allowed_origin(mut self, origin: Locality) -> Self {
         self.subscribers_origin = origin;
         self
     }
     /// Restrict the matching queryables that will receive queries from this [`Session`]
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:404]: Locality should not be configurable at session level but only per declare_subscriber and declare_queryable-basis.

File: zenoh/src/lib.rs [hunk 11]
```diff
@@ -403,6 +410,7 @@ impl InitBuilder {
 
     /// Restrict the matching queries that will be receive by the [`Queryables`](crate::queryable::Queryable)
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/lib.rs:413]: Locality should not be configurable at session level but only per declare_subscriber and declare_queryable-basis.

File: zenoh/src/publication.rs [hunk 12]
```diff
@@ -100,6 +100,7 @@ impl PutBuilder<'_, '_> {
 
     /// Restrict the matching subscribers that will receive the published data
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/publication.rs:103]: `Locality` should be only on subscriber and queryable side for the time being at least.

File: zenoh/src/publication.rs [hunk 13]
```diff
@@ -241,6 +242,7 @@ impl<'a> Publisher<'a> {
...
```
Comment (Mallets) [inline zenoh/src/publication.rs:245]: `Locality` should be only on subscriber and queryable side for the time being at least.

File: zenoh/src/publication.rs [hunk 14]
```diff
@@ -536,6 +538,7 @@ impl<'a, 'b> PublisherBuilder<'a, 'b> {
...
```
Comment (Mallets) [inline zenoh/src/publication.rs:541]: `Locality` should be only on subscriber and queryable side for the time being at least.

File: zenoh/src/query.rs [hunk 15]
```diff
@@ -280,6 +280,7 @@ impl<'a, 'b, Handler> GetBuilder<'a, 'b, Handler> {
 
     /// Restrict the matching queryables that will receive the query
     /// to the ones that have the given [`Locality`](crate::prelude::Locality).
+    #[cfg(feature = "unstable")]
```
Comment (Mallets) [inline zenoh/src/query.rs:283]: `Locality` should be only on subscriber and queryable side for the time being at least.

**Review by Mallets — CHANGES_REQUESTED** [review]

**Comment by OlivierHecart** [comment]
> I believe that also `Session::init` parameters should be marked `unstable`.
> 
> See here:
> 
> https://github.com/eclipse-zenoh/zenoh/blob/4edb8590aa4eec3188f519d69635c01e6ad6a5bf/zenoh/src/session.rs#L324

`Session::init` is now private (`pub(crate)`). No need to mark it as unstable.

**Review by Mallets — APPROVED** [review]

**Pull request merged**